src/Plugin/MembershipProviderManager.php

```diff
@@ -1,8 +1,10 @@
 <?php
 
-namespace Drupal\membership_provider\Plugin;
+namespace Drupal\membership\Plugin;
 
-use Drupal\Core\Entity\EntityFieldManager;
+use Drupal\Core\Config\Entity\Query\QueryFactory;
```

Comment:
Service container is injecting a Drupal\Core\Entity\Query\QueryFactory, not a Drupal\Core\Config\Entity\Query\QueryFactory